docs/assignment2.md

```diff
@@ -1 +1,322 @@
+* _getInstance(): JsonIO_
+    * returns the singleton json object.
+* _writeToJson(pathToJson:String, object :Object): void_
+    * serializes an object.
+* _loadFromJson(pathToJson:String): Snippet_
+    * deserializes a Snippet.
+* _onException(e:Exception): void_
+    * handles exceptions like file not found or bad data.
+#### Associations
+* _JsonIO < Main_
+    * already described
+* _JsonIO < Snippet_
+    * The snippet class uses the JsonIO class to convert snippet objects to json, to store snippet objects.
+* _JsonIO < SnippetManager_
+    * The snippet manager class uses the JsonIO class to convert json to objects, to retrieve snippet objects.
+
+### class: Main
+ 
+#### Attributes
+* _snippetManager: SnippetManager_
+    * the snippet manager used by the current instance of the app.
+* _cliUI: CliUI_
+    * the cliUI used by the current instance of the app.
+#### Operations
+* _main(args:String[*]): void_
+    * Instantiates SnippetManager, CliUI and JsonIO. After this, CliUI handles the flow of the app.
+#### Associations
+* _Main > JsonIO_
+    * instantiates.
+* _Main > SnippetManager_
+    * instantiates.
+* _Main > CliUI_
+    * instantiates.
+    
+## Object diagrams								
+Author(s): Yael Goede
+
+![snapshot Diagram](snapshotDiagram.png)
+In the diagram above a snapshot of the system is shown where the database class is filled with 4 snippets. All snippets have a unique id which makes them unique from each other. Further more all fields are filled with the required information, such as the language, tags and content field. 
+The content field contains the actual code used as snippet by the user. The language field specifies the programming language which is valid for the content field. This field enables features such as the text highlighting feature, which is language specific.
+
+## State machine diagrams									
+Author(s): `Björn Keyser`
+
+### Class CLiUI
+![snapshot Diagram](cli_stateDiagram.jpeg)
+
+#### States
+- Initial State: Here we display the menu and wait for the user input. In event of valid user input, one of the 6 functionalities is executed.
+- View all snippets: Lists all snippets and their attributes (title, content, tags, language, metadata)
+- Prompt for details [for creation of snippet]: In event of choosing to create a snippet, the user is prompted for a title, the language and tags. A snippet is then created and a editor is opened with no content. 
+- Editor mode: The editor loads content if the editor opened a existing snippet, and a blank page otherwise. On exit, the user is prompted to either save the file or not.
+- Prompt for ID: In case of deleting or editing a snippet, the user is prompted for the ID of the snippet to be deleted/edited. If the ID is valid the snippet is deleted/opened in the editor.
+- View found snippets: In event of filtering, the user can specify parameters for filtering on title name, content or tags and the found snippets are displayed.
+- Quit: quit the program
+
```

Comment:
-State machine diagrams are done for each stateful class of the class diagram
-Which class does your state machine represent?
-For example, The snippet could have states for: Creation, Update, Deletion etc
-States are nouns and activities are verbs
-This is an activity diagram that's trying to represent your entire system, which isn't what we want here